### ISSUE TITLE
Fatal Lua Error: NullReferenceException in 07: Sunken Treasures

### ISSUE THREAD

**Issue by Mailaender** [OpenRA/OpenRA#17711]
Happens right on the start of the mission.

```
Fatal Lua Error: [string "BindingSupport.lua"]:30: Uncaught CLR exception at Lua->CLR boundary: System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation. ---> System.NullReferenceException: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.Traits.ArmyUnit.<.ctor>b__11_2 (OpenRA.Mods.Common.Traits.ProductionQueueInfo q) [0x00000] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs:184 
  at System.Linq.Enumerable+WhereSelectEnumerableIterator`2[TSource,TResult].MoveNext () [0x00037] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/external/corefx/src/System.Linq/src/System/Linq/Where.cs:699 
  at OpenRA.Exts.CompareBy[T,U] (System.Collections.Generic.IEnumerable`1[T] ts, System.Func`2[T,TResult] selector, System.Int32 modifier, System.Boolean throws) [0x0000d] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Exts.cs:221 
  at OpenRA.Exts.MinByOrDefault[T,U] (System.Collections.Generic.IEnumerable`1[T] ts, System.Func`2[T,TResult] selector) [0x00000] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Exts.cs:206 
  at OpenRA.Mods.Common.Traits.ArmyUnit..ctor (OpenRA.ActorInfo actorInfo, OpenRA.Player owner) [0x00088] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs:184 
  at OpenRA.Mods.Common.Traits.PlayerStatistics+<>c__DisplayClass24_0.<.ctor>b__0 (System.String name) [0x00000] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs:75 
  at OpenRA.Exts.GetOrAdd[K,V] (System.Collections.Generic.Dictionary`2[TKey,TValue] d, K k, System.Func`2[T,TResult] createFn) [0x0000b] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Exts.cs:120 
  at OpenRA.Primitives.Cache`2[T,U].get_Item (T key) [0x00000] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Primitives/Cache.cs:36 
  at OpenRA.Mods.Common.Traits.UpdatesPlayerStatistics.OpenRA.Traits.INotifyCreated.Created (OpenRA.Actor self) [0x00031] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/Player/PlayerStatistics.cs:267 
  at OpenRA.Actor.Created () [0x0001c] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Actor.cs:147 
  at OpenRA.World.CreateActor (System.Boolean addToWorld, System.String name, OpenRA.Primitives.TypeDictionary initDict) [0x00009] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/World.cs:340 
  at OpenRA.Mods.Common.Scripting.ReinforcementsGlobal.CreateActor (OpenRA.Player owner, System.String actorType, System.Boolean addToWorld, System.Nullable`1[T] entryLocation, System.Nullable`1[T] nextLocation) [0x00103] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs:60 
  at OpenRA.Mods.Common.Scripting.ReinforcementsGlobal.ReinforceWithTransport (OpenRA.Player owner, System.String actorType, System.String[] cargoTypes, OpenRA.CPos[] entryPath, OpenRA.CPos[] exitPath, Eluant.LuaFunction actionFunc, Eluant.LuaFunction exitFunc, System.Int32 dropRange) [0x0000d] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Scripting/Global/ReinforcementsGlobal.cs:125 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:395 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007e] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:409 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/external/corefx/src/Common/src/CoreLib/System/Reflection/MethodBase.cs:53 
  at OpenRA.Scripting.ScriptMemberWrapper.Invoke (Eluant.LuaVararg args) [0x000f0] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Scripting/ScriptMemberWrapper.cs:75 
  at (wrapper managed-to-native) System.Reflection.RuntimeMethodInfo.InternalInvoke(System.Reflection.RuntimeMethodInfo,object,object[],System.Exception&)
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0006a] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:395 
   --- End of inner exception stack trace ---
  at System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) [0x0007e] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/mcs/class/corlib/System.Reflection/RuntimeMethodInfo.cs:409 
  at System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) [0x00000] in /home/abuild/rpmbuild/BUILD/mono-6.4.0.198/external/corefx/src/Common/src/CoreLib/System/Reflection/MethodBase.cs:53 
  at Eluant.LuaRuntime+MethodWrapper.Invoke (System.Object[] parms) [0x0000d] in <c8bc0b4612b543268b8307855c69585a>:0 
  at Eluant.LuaRuntime.MakeManagedCall (System.IntPtr state, Eluant.LuaRuntime+MethodWrapper wrapper) [0x0047d] in <c8bc0b4612b543268b8307855c69585a>:0 
  at OpenRA.Scripting.ScriptContext.FatalError (System.String message) [0x00000] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.Mods.Common.Scripting.TriggerGlobal+<>c__DisplayClass2_0.<AfterDelay>b__0 () [0x0002b] in <2dad04dc7f7648df8c4e3f1f7ae6ffd9>:0 
  at OpenRA.Effects.DelayedAction.<Tick>b__3_0 (OpenRA.World w) [0x00007] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.World.Tick () [0x0019e] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.Game.LogicTick () [0x0003e] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.Game.Loop () [0x000f1] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.Game.Run () [0x0003c] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
  at OpenRA.Program.Main (System.String[] args) [0x00044] in <cd1fa4b708d94ec9b3f6f4a9c199feef>:0 
Exception of type `System.NullReferenceException`: Object reference not set to an instance of an object
  at OpenRA.Mods.Common.Traits.Mobile.set_CurrentMovementTypes (OpenRA.Mods.Common.Traits.MovementType value) [0x00052] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/Mobile.cs:187 
  at OpenRA.Mods.Common.Traits.Mobile.UpdateMovement (OpenRA.Actor self) [0x0004f] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/Mobile.cs:324 
  at OpenRA.Mods.Common.Traits.Mobile.OpenRA.Traits.ITick.Tick (OpenRA.Actor self) [0x00000] in /home/matthias/Entwicklung/OpenRA/OpenRA.Mods.Common/Traits/Mobile.cs:309 
  at OpenRA.World+<>c.<Tick>b__110_0 (OpenRA.TraitPair`1[T] x) [0x00000] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/World.cs:464 
  at OpenRA.WorldUtils.DoTimed[T] (System.Collections.Generic.IEnumerable`1[T] e, System.Action`1[T] a, System.String text) [0x00015] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/WorldUtils.cs:78 
  at OpenRA.World.Tick () [0x00151] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/World.cs:464 
  at OpenRA.Game.InnerLogicTick (OpenRA.Network.OrderManager orderManager) [0x001bc] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:602 
  at OpenRA.Game.LogicTick () [0x0003e] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:629 
  at OpenRA.Game.Loop () [0x000f1] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:793 
  at OpenRA.Game.Run () [0x0003c] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:834 
  at OpenRA.Game.InitializeAndRun (System.String[] args) [0x00010] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Game.cs:263 
  at OpenRA.Program.Main (System.String[] args) [0x00044] in /home/matthias/Entwicklung/OpenRA/OpenRA.Game/Support/Program.cs:37 
```

**Comment by abcdefg30** [comment]
This was fixed by #17692 I presume.

**Comment by Mailaender** [comment]
Yes, it does not happen on latest bleed.
